### ISSUE TITLE
executing makefile twice in succession outputs 8xp even if linking fails

### ISSUE THREAD

**Issue by elfprince13** [CE-Programming/toolchain#7]
The resulting 8xp is however still missing the symbols from the first attempt.
See, for example:

```
C:\Scanline-Rendering\ScanLineRender\ScanLineRender>..\toolchain\CEdev\bin\make.exe
ael.c
DEBUGCONFIG.H   (12,8) :        WARNING (3) Illegal redefinition of macro "NDEBUG"
ael.src
edge.c
DEBUGCONFIG.H   (12,8) :        WARNING (3) Illegal redefinition of macro "NDEBUG"
edge.src
point.c
point.src
primitive.c
DEBUGCONFIG.H   (12,8) :        WARNING (3) Illegal redefinition of macro "NDEBUG"
primitive.src
scanlinerenderer.c
DEBUGCONFIG.H   (12,8) :        WARNING (3) Illegal redefinition of macro "NDEBUG"
scanlinerenderer.src
transformation.c
transformation.src
main.c
DEBUGCONFIG.H   (12,8) :        WARNING (3) Illegal redefinition of macro "NDEBUG"
main.src
Linking...
WARNING (724) --> Symbol _RBMapPut (ScanlineRenderer.obj) is not defined.
WARNING (724) --> Symbol _RBDelete (ScanlineRenderer.obj) is not defined.
WARNING (724) --> Symbol _RBMapRemove (ScanlineRenderer.obj) is not defined.
WARNING (724) --> Symbol _RBTreeDestroy (ScanlineRenderer.obj) is not defined.
WARNING (724) --> Symbol _RBMapNodeAlloc (ScanlineRenderer.obj) is not defined.
WARNING (724) --> Symbol _mergeSort (AEL.obj) is not defined.
WARNING (724) --> Symbol _RBSetContains (ScanlineRenderer.obj) is not defined.
WARNING (724) --> Symbol _RBTreeInsert (ScanlineRenderer.obj) is not defined.
WARNING (724) --> Symbol _RBNodeAlloc (ScanlineRenderer.obj) is not defined.
WARNING (724) --> Symbol _SafeMalloc (AEL.obj) is not defined.
WARNING (724) --> Symbol _RBSetAdd (ScanlineRenderer.obj) is not defined.
WARNING (724) --> Symbol _RBTreeMapInit (ScanlineRenderer.obj) is not defined.
WARNING (724) --> Symbol _RBExactQuery (ScanlineRenderer.obj) is not defined.
WARNING (724) --> Symbol _freeLink (AEL.obj) is not defined.
WARNING (724) --> Symbol _removeLink (AEL.obj) is not defined.
WARNING (724) --> Symbol _linkFront (AEL.obj) is not defined.
WARNING (724) --> Symbol _RBTreeInit (ScanlineRenderer.obj) is not defined.
WARNING (724) --> Symbol _RBTreeClear (ScanlineRenderer.obj) is not defined.
WARNING (724) --> Symbol _TreeSuccessor (AEL.obj) is not defined.

SPACE ALLOCATION:
=================

Group: group->name                          Base         Top      Size      Used    Unused
------------------------------------ ----------- ----------- --------- --------- ---------
Space: RAM                              00D031F6    00D1DCF9    65801H     352FH    622D2H
                                                            (   415745     13615    402130)

Space                     Base         Top      Size      Used    Unused
------------------ ----------- ----------- --------- --------- ---------
RAM                   D:D031F6    D:D1DCF9    65801H     352FH    622D2H
                                          (   415745     13615    402130)

ERROR: Object file(s) deleted because of option unresolved=fatal.
Consult scanln.map for more information.
make: *** [scanln.hex] Error -1

C:\Scanline-Rendering\ScanLineRender\ScanLineRender>..\toolchain\CEdev\bin\make.exe

Input File: scanln.hex
Output File: scanln.8xp
Output Calculator Name: SCANLN
Mark archived: No
Success!

Program Size: 13450 bytes
```

**Comment by MateoConLechuga** [comment]
Hm, the .hex file should be deleted if the linker fails

**Comment by elfprince13** [comment]
Can we use || in Windows make to do a failure condition? I assume so since
using && worked for the cd in the .o targets.

On Thursday, February 4, 2016, Matt Waltz <notifications@github.com> wrote:

> Hm, the .hex file should be deleted if the linker fails
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/CE-Programming/toolchain/issues/7#issuecomment-180121133>
> .
>


-- 
~Thomas

**Comment by MateoConLechuga** [comment]
Yeah, that should be a possibility

**Comment by MateoConLechuga** [comment]
Many thanks to jacobly, fixed with commit 2c76fa4
